### PR TITLE
Modified arbiter to get valid GCS access token on windows.

### PR DESCRIPTION
**Pull request by pravinshinde825**
- `Libexecinfo` and `Libunwind` are not for windows, Modified to ignore them on windows.
- `ShLwApi` is already in path.
- `utcOffsetSeconds` throws an null pointer exception on windows. Which causes the failure in retriving the access token from GCS. Added new method for windows named `utcToLocalOffsetSeconds` which will return the UTC to Local time difference. After this arbiter can retrieve a valid access token from 
GCS.

### CODE REVIEW

**Comment by nicolas-chaulet** [comment]
Could we keep this PR out of master? Maybe create a branch that is like helix-master. If we merge that into master then synching with upstream and submitting PRs will become a bit challenging I think...

**Comment by pravinshinde825** [comment]
@nicolas-chaulet , Changed base branch to helix-master.

File: cmake/execinfo.cmake
Comment (nicolas-chaulet) [inline cmake/execinfo.cmake:4]: This is strange you need that because this works for PDAL main repo. Is it an MSCV 2017 issue?

File: cmake/unwind.cmake
Comment (nicolas-chaulet) [inline cmake/unwind.cmake:4]: Same as above

File: vendor/arbiter/CMakeLists.txt
Comment (nicolas-chaulet) [inline vendor/arbiter/CMakeLists.txt:24]: This work in the official repo, why do we need to change it?

File: vendor/arbiter/arbiter.cpp
Comment (nicolas-chaulet) [inline vendor/arbiter/arbiter.cpp]: This code is part the arbiter repo:
https://github.com/connormanning/arbiter
Couple of problems with modifying our version:
- we will have to maintain a fork
- I don't thiink you solve the root of the problem
I think you should submit a PR to the arbiter tepo first, get feedback from the maintainer with the right fix and then we can integrate here. @smileprem any objection?

**Review by nicolas-chaulet — CHANGES_REQUESTED** [review]

File: vendor/arbiter/arbiter.cpp
Comment (premkumar-masilamani) [reply vendor/arbiter/arbiter.cpp]: That's the right approach. Let's do that. However, should we wait for the PR to be accepted to move on with the current code changes?.

Can we fork the aribiter, make changes and submit a PR, while that is in process, use our arbiter codebase for `helix-master` branch?. I know the code will not be merged to master until then, but we can finish the refactoring and complete the testing?.

I am trying to save some time here. Finally there will be a merge from `helix-master` to `master` anyway.

File: cmake/execinfo.cmake
Comment (pravinshinde825) [reply cmake/execinfo.cmake:4]: Yes it is related to MSVC 2017.

File: vendor/arbiter/CMakeLists.txt
Comment (pravinshinde825) [reply vendor/arbiter/CMakeLists.txt:24]: I tried this with official repo as well it didn't worked.
It cannot find shlwapi lib.

File: vendor/arbiter/arbiter.cpp
Comment (nicolas-chaulet) [reply vendor/arbiter/arbiter.cpp]: @pravinshinde825 could you please submit a PR in arbiter in order to start the conversation? I'll merge this straight after. Regarding the cmake seetings above, could you please raise an issue on PDAL repo and ask what's up? It is supposed too be cross platfrom and they use MSVC 2017. SO no eason it does not work for us. It is really important we keep close to the original repos, they are moving quite fast. Thx!

Comment (pravinshinde825) [reply vendor/arbiter/arbiter.cpp]: @nicolas-chaulet Sure I'll check arbiter repo and create PR there. But I think before creating PR we should raise an issue to arbiter repo. What do you think?

**Comment by nicolas-chaulet** [comment]
You can simply explain the bug in the pr itself. You can also Try to
compile the arbiter on your machine and run the tests and explain which one
fails, there should be one for this functionality

**Comment by pravinshinde825** [comment]
@nicolas-chaulet @smileprem , I found the root cause of this issue.
The issue is `std::mktime()` returns -1, Since the time in `std::tm` cannot be represented as a `std::time_t` object. 
The solution to this problem can be: Use `std::time(nullptr)` to get current time in `std::time_t`.
Looking forward to implement the same. Once done will submit PR to arbiter.

**Comment by nicolas-chaulet** [comment]
Nice!! Sounds reasonable to me

**Comment by pravinshinde825** [comment]
I've created PR to arbiter repo : https://github.com/connormanning/arbiter/pull/28 .
It is just one line change.

**Comment by nicolas-chaulet** [comment]
Well done! If you update this pr I am happy to merge it into helix-master

**Comment by premkumar-masilamani** [comment]
> Well done! If you update this pr I am happy to merge it into helix-master
> […](#)

What about the questions related to MSCV 2017?. They are still not resolved.

**Review by nicolas-chaulet — APPROVED** [review]

**Pull request merged**
